python/vyos/qos/base.py

```diff
@@ -331,15 +331,6 @@ def update(self, config, direction, priority=None):
                         filter_cmd += f' flowid {self._parent:x}:{cls:x}'
                         self._cmd(filter_cmd)
 
-                else:
-
-                    filter_cmd += ' basic'
-
-                    cls = int(cls)
-                    filter_cmd += f' flowid {self._parent:x}:{cls:x}'
```

Comment:
It was added to fix previous issues with multiple matches in https://github.com/vyos/vyos-1x/commit/4917d7ab4601904bb19e9e2c8ea3e4bf7bc7521d

Check please that multiple matches are not affected

```
set qos interface eth0 egress 'test'
set qos policy shaper test bandwidth '300mbit'
set qos policy shaper test class 23 bandwidth '150mbit'
set qos policy shaper test class 23 match 10 ip protocol 'tcp'
set qos policy shaper test class 23 match 20 ip protocol 'udp'
set qos policy shaper test default bandwidth '20mbit'
set qos policy shaper test default queue-type 'fair-queue'
```

More details in https://vyos.dev/T5302

Reply:
@sever-sever everithing works good without any error

```
vyos@vyos# commit
[ qos ]
DEBUG/QoS: tc qdisc replace dev eth0 root handle 1: htb r2q 188 default 18
DEBUG/QoS: tc class replace dev eth0 parent 1: classid 1:1 htb rate 300000000
DEBUG/QoS: tc class replace dev eth0 parent 1:1 classid 1:17 htb rate 150000000 burst 15k quantum 1514
DEBUG/QoS: tc qdisc replace dev eth0 parent 1:17 sfq
DEBUG/QoS: tc class replace dev eth0 parent 1:1 classid 1:18 htb rate 20000000 burst 15k quantum 1514 prio 20
DEBUG/QoS: tc qdisc replace dev eth0 parent 1:18 sfq
{'bandwidth': '300mbit',
 'class': {'23': {'bandwidth': '150mbit',
                  'burst': '15k',
                  'codel_quantum': '1514',
                  'flows': '1024',
                  'interval': '100',
                  'match': {'10': {'ip': {'protocol': 'tcp'}},
                            '20': {'ip': {'protocol': 'udp'}}},
                  'queue_type': 'fq-codel',
                  'target': '5'}},
 'default': {'bandwidth': '20mbit',
             'burst': '15k',
             'codel_quantum': '1514',
             'flows': '1024',
             'interval': '100',
             'priority': '20',
             'queue_type': 'fair-queue',
             'target': '5'}}
DEBUG/QoS: tc qdisc replace dev eth0 parent 1:17 fq_codel quantum 1514 flows 1024 interval 100 interval 100 target 5 noecn
DEBUG/QoS: tc filter add dev eth0 parent 1: protocol all prio 1 u32 match ip protocol 6 0xff flowid 1:17
DEBUG/QoS: tc filter add dev eth0 parent 1: protocol all prio 2 u32 match ip protocol 17 0xff flowid 1:17
DEBUG/QoS: tc filter add dev eth0 parent 1: protocol all prio 2 u32 match ip protocol 17 0xff flowid 1:17 action police rate 150000000 burst 15k flowid 1:17
DEBUG/QoS: tc qdisc replace dev eth0 parent 1:18 sfq

[edit]
```